addons/account/models/account_bank_statement.py

```diff
@@ -1249,15 +1249,18 @@ def reconcile(self, lines_vals_list, to_check=False, allow_partial=False):
     # -------------------------------------------------------------------------
 
     def _find_or_create_bank_account(self):
-        bank_account = self.env['res.partner.bank'].search(
-            [('company_id', '=', self.company_id.id), ('acc_number', '=', self.account_number)])
+        bank_account = self.env['res.partner.bank'].search([
+            ('company_id', '=', self.company_id.id),
+            ('acc_number', '=', self.account_number),
+            ('partner_id', '=', self.partner_id.id),
+        ])
         if not bank_account:
             bank_account = self.env['res.partner.bank'].create({
                 'acc_number': self.account_number,
                 'partner_id': self.partner_id.id,
                 'company_id': self.company_id.id,
             })
-        return bank_account
+        return bank_account if not bank_account.is_credit_collector else None
```

Comment:
Aren't you suppose to avoir the creation of a new bank account in that case? It looks weird to me to create a bank account on background but not used and completely invisibile for the user

Reply:
I don't know, a bank account is not a credit collector by default, so the newly created one won't be invisible.